examples/ExportSystemsToCSV/exportSystemsToCSV.go

```diff
@@ -40,8 +82,34 @@ func main() {
 			system.AmazonInstanceID, system.Os, system.Version, system.AgentVersion, system.Created,
 			system.LastContact}
 
-		for _, tag := range system.Tags {
-			outLine = append(outLine, tag.Name)
+		if isGroups {
```

Comment:
This level of nesting probably merits being split out into a couple functions. Notably, the loop bodies and the if/else blocks all seem like good candidates. Something like
```
if isGroups {
	groupsNames := getGroupsNames(<whatever params>)
	outLine = append(outLine, groupsNames...)
} else {
	tagNames := getTagsNames(<whatever params>)
	outLine = append(outLine, tagNames...)
}
```

And then inside getGroupsNames, you'd have a separate function that you call in the loop that does the `for _, graph := range graphs...` chunk, etc.

Reply:
I agree with moving the groups loop code into a separate function but the tags code is just a one-liner for loop, I think it's making a function for that is a bit overkill.

Reply:
Sure, that sounds reasonable.